### PR TITLE
Split worker pools docs into multiple sections

### PR DESCRIPTION
**Pull request by adamconnelly**
# Description of the change

The worker pools page has become pretty huge since we added the docs about Kubernetes workers. I've tried to split it up, with a main page explaining the overall concept, and sub-pages that explain how to deploy workers using either Docker or Kubernetes.

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [x] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/worker-pools/docker-based-workers.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+  --capabilities "CAPABILITY_NAMED_IAM"
+```
+
+## Terraform Modules
+
+Our public [AWS](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2){: rel="nofollow"}, [Azure](https://github.com/spacelift-io/terraform-azure-spacelift-workerpool){: rel="nofollow"} and [GCP](https://github.com/spacelift-io/terraform-google-spacelift-workerpool){: rel="nofollow"} Terraform modules are not currently compatible with self-hosting.
+
+{% endif %}
+
+## Passing metadata tags
```
Comment (adamconnelly) [inline docs/concepts/worker-pools/docker-based-workers.md:309]: Note to self - the frontend currently references this section, so we'll need to also update it to point at a different page.

Comment (adamconnelly) [reply docs/concepts/worker-pools/docker-based-workers.md]: I've actually moved this back to the main worker pools page, so this isn't required anymore.

**Comment by peterdeme** [comment]
@adamconnelly I'm reading through the docs. Just one minor fix in the `Docker-based Workers`:

> You can do this by draining all the workers in the UI, then terminating the instances in your cloud provider. That should automatically start new instances with the latest launcher version.

We now have a cycle button thank god, so we can rephrase this a bit. 🙂 Not just that, but this section is duplicated. If you scroll down, there is another one with a similar text. This one isn't needed:

> Since the Launcher is downloaded during instance startup, it is recommended to recycle the worker pool every once in a while to ensure that it is up to date. You don't want to miss out on the latest features and bug fixes! You can do this by draining all the workers one-by-one in the UI, then terminating the instances in your cloud provider.

**Review by peterdeme — DISMISSED** [review]

**Review by peterdeme — APPROVED** [review]

**Review by eliecharra — APPROVED** [review]
Makes perfect sense, thanks for doing that 👌🏻

**Comment by adamconnelly** [comment]
Good catch @peterdeme.

**Pull request merged**